service/controller/chart/resource/release/create.go

```diff
@@ -112,7 +112,16 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 		r.logger.Debugf(ctx, "canceling resource")
 		return nil
 	}
-	if helmclient.IsValidationFailedError(err) {
+	if helmclient.IsResourceAlreadyExists(err) {
```

Comment:
Even though the original issue was from update events, we also need to care for create.go as well.